### PR TITLE
[SPARK-47526][BUILD] Upgrade `netty` to 4.1.108.Final and `netty-tcnative` to 2.0.65.Final

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade:
- `netty` from `4.1.107.Final` to `4.1.108.Final`.
- `netty-tcnative` from `2.0.62.Final` to `2.0.65.Final`.

### Why are the changes needed?
- `netty`
1.The release notes: https://netty.io/news/2024/03/21/4-1-108-Final.html
2.To bring some bug fixes, eg:
Epoll: Fix possible Classloader deadlock caused by loading class via JNI ([#13879](https://github.com/netty/netty/issues/13879))
  
- `netty-tcnative`
2.0.62.Final VS 2.0.65.Final
https://github.com/netty/netty-tcnative/compare/netty-tcnative-parent-2.0.62.Final...netty-tcnative-parent-2.0.65.Final

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: dev/deps/spark-deps-hadoop-3-hive-2.3 [hunk 1]
```diff
@@ -16,10 +16,10 @@ antlr4-runtime/4.13.1//antlr4-runtime-4.13.1.jar
...
 arpack/3.0.3//arpack-3.0.3.jar
 arpack_combined_all/0.1//arpack_combined_all-0.1.jar
-arrow-format/15.0.0//arrow-format-15.0.0.jar
-arrow-memory-core/15.0.0//arrow-memory-core-15.0.0.jar
-arrow-memory-netty/15.0.0//arrow-memory-netty-15.0.0.jar
-arrow-vector/15.0.0//arrow-vector-15.0.0.jar
+arrow-format/15.0.2//arrow-format-15.0.2.jar
+arrow-memory-core/15.0.2//arrow-memory-core-15.0.2.jar
+arrow-memory-netty/15.0.2//arrow-memory-netty-15.0.2.jar
+arrow-vector/15.0.2//arrow-vector-15.0.2.jar
```
Comment (dongjoon-hyun) [inline dev/deps/spark-deps-hadoop-3-hive-2.3:22]: To @panbingkun , we had better handle `arrow` dependency first independently.

Comment (panbingkun) [reply dev/deps/spark-deps-hadoop-3-hive-2.3:22]: Okay, let me separate them.  Just a moment.

Comment (panbingkun) [reply dev/deps/spark-deps-hadoop-3-hive-2.3]: Done

**Comment by dongjoon-hyun** [comment]
#45682 is merged.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @panbingkun .
Merged to master.

**Pull request closed without merging**